### PR TITLE
[docs] Add citation notebooks

### LINKED ISSUE #615: Add notebook that demonstrates how to obtain a list of citations for a given Census slice
Blocked by #559 and #560.

Create python and R notebooks that demonstrate:

1. For an `obs` slice, obtain the corresponding rows from the  `datasets` data frame and obtain a list of unique citations.
2. For an `ExperimentAxisQuery`, obtain the corresponding rows from the  `datasets` data frame and obtain a list of unique citations.

### PR DESCRIPTION
**Pull request by pablo-gar**
fixes #615

### CODE REVIEW

File: api/python/notebooks/api_demo/census_citation_generation.ipynb [hunk 1]
```diff
@@ -0,0 +1,331 @@
...
+    "# Generating citations for Census slices\n",
+    "\n",
+    "This notebook demonstrates how to generate a citation string for all datasets contained in a Census slice.\n",
+    "\n",
+    "**Contents**\n",
+    "\n",
+    "1. Requirements\n",
+    "1. Generating citation strings\n",
+    "   1. Via cell metadata query\n",
+    "   1. Via a AnnData query \n",
```
Comment (ebezzi) [inline api/python/notebooks/api_demo/census_citation_generation.ipynb:17]:
```suggestion
    "   1. Via an AnnData query \n",
```

File: api/python/notebooks/api_demo/census_citation_generation.ipynb [hunk 2]
```diff
@@ -0,0 +1,331 @@
...
+    "    column_names={\"obs\": [\"dataset_id\", \"cell_type\"]},\n",
+    ")\n",
+    "\n",
+    "# Get a citation string for the slice\n",
+    "slice_datasets = datasets[datasets[\"dataset_id\"].isin(adata.obs[\"dataset_id\"])]\n",
+    "print(*slice_datasets[\"citation\"], sep=\"\\n\\n\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
```
Comment (ebezzi) [inline api/python/notebooks/api_demo/census_citation_generation.ipynb:271]: This cell should probably move after the `.shape` one.

**Review by ebezzi — APPROVED** [review]

File: api/python/notebooks/api_demo/census_citation_generation.ipynb (hunk 2 above)
Comment (pablo-gar) [reply api/python/notebooks/api_demo/census_citation_generation.ipynb:271]: This one was for testing, I removed it.

**Review by atolopko-czi — APPROVED** [review]
LGTM w/one minor concern: If the census slice has multiple datasets and they all have the same publication, should the notebook do a "unique" on this set before printing?

**Comment by pablo-gar** [comment]
@atolopko-czi the code guarantees no duplicates will be shown as is.

**Pull request merged**